Sources/Sentry/SentryTransaction.m

```diff
@@ -14,10 +16,18 @@ - (instancetype)initWithTrace:(id<SentrySpan>)trace children:(NSArray<id<SentryS
         _trace = trace;
         _spans = children;
         self.type = SentryEnvelopeItemTypeTransaction;
+        measurements = [NSMutableDictionary new];
     }
     return self;
 }
 
+- (void)setMeasurementValue:(id)value forKey:(NSString *)key
+{
+    @synchronized(measurements) {
```

Comment:
Transaction can't (shouldn't) be accessed concurrentlyt so we should need this synchronized here.

Don't we have the Tracer class that creates an instance of `SentryTransaction` and captures that? If tthat's still the case no other thread has a reference to this instance.

Same idea with `SentryEvent` which isn't thread-safe.